From a62a1e1d6273d1b2021b072603ee9ffc86afc5c5 Mon Sep 17 00:00:00 2001
From: Ulysses Souza <ulyssessouza@gmail.com>
Date: Thu, 16 Apr 2020 17:04:40 +0200
Subject: [PATCH] Add "distro" package

This package implements the method 'platform.linux_distribution' removed in
Python 3.8

Signed-off-by: Ulysses Souza <ulyssessouza@gmail.com>
---
 compose/cli/utils.py | 3 ++-
 requirements.txt     | 1 +
 setup.py             | 1 +
 3 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/compose/cli/utils.py b/compose/cli/utils.py
index 931487a6cd..40bef16520 100644
--- a/compose/cli/utils.py
+++ b/compose/cli/utils.py
@@ -9,6 +9,7 @@
 import subprocess
 import sys
 
+import distro
 import docker
 import six
 
@@ -73,7 +74,7 @@ def is_mac():
 
 
 def is_ubuntu():
-    return platform.system() == 'Linux' and platform.linux_distribution()[0] == 'Ubuntu'
+    return platform.system() == 'Linux' and distro.linux_distribution()[0] == 'Ubuntu'
 
 
 def is_windows():
diff --git a/requirements.txt b/requirements.txt
index 76556d6..edc30a7 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -4,6 +4,7 @@ cached-property==1.5.1
 certifi==2019.11.28
 chardet==3.0.4
 colorama==0.4.3; sys_platform == 'win32'
++distro==1.5.0
 docker==4.1.0
 docker-pycreds==0.4.0
 dockerpty==0.4.1
diff --git a/setup.py b/setup.py
index bca578e110..cf273ed548 100644
--- a/setup.py
+++ b/setup.py
@@ -36,6 +36,7 @@ def find_version(*file_paths):
     'requests >= 2.20.0, < 3',
     'texttable >= 0.9.0, < 2',
     'websocket-client >= 0.32.0, < 1',
+    'distro >= 1.5.0, < 2',
     'docker[ssh] >= 3.7.0, < 5',
     'dockerpty >= 0.4.1, < 1',
     'six >= 1.3.0, < 2',
